### PR TITLE
Fix usage of closures in validation

### PR DESCRIPTION
**Pull request by wimski**
- [x] Added or updated tests
- [x] Updated CHANGELOG.md

Solves this issue https://github.com/nuwave/lighthouse/pull/1824#issuecomment-826706848

### CODE REVIEW

**Comment by wimski** [comment]
@spawnia

File: tests/Utils/Validators/FooClosureValidator.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
```
Comment (spawnia) [inline tests/Utils/Validators/FooClosureValidator.php:3]: I plan to switch to this in the near future, but want to keep the codebase consistent for now.

```suggestion
```

File: tests/Utils/Validators/FooClosureValidator.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+
```
Comment (spawnia) [inline tests/Utils/Validators/FooClosureValidator.php:3]:
```suggestion
```

**Comment by spawnia** [comment]
Thank you, nicely done

**Comment by spawnia** [comment]
For the record, https://github.com/nuwave/lighthouse/pull/1739 is what broke validation rules with closures.

**Pull request merged**